Adding API to get ovs controller state

vareti opened the pull request:
This api can be used to determine the state of the controller. It relies on Open_vSwitch `Controller` table  `role` column.

Commit:
api to get ovs controller state

pdellaert reviewed (commented):
@vareti, are we only checking one controller? Not the different scenarios where we have one or two controllers, and can provide controller state for each?

I understand that only checking the master is enough to determine if there is a controller connection (because, if two controllers are configured, and only one is connected, that one automatically is considered the master), just wondering if it could be interesting to check state for both of there are two. (or perhaps support that in the future)

Another point: The test really is pretty limited (as it doesn't test different connectivity states and requires OVS running), not that this is a blocker, but we might want to see if we can run tests without the need for OVS running (mock it) and test all possible returns and errors.

pdellaert reviewed (dismissed):
LGTM for limited required functionality, check previous comment to think about improvements

Commit:
more detailed unit test, add table mock to be used by tests

vareti commented:
@pdellaert It was intentional to just look for one master controller. There could be more cases like Dual VTEP VRS where there could be two master controller or we can decide per controller but I am thinking that can be part of future enhancement.

I added a mock for ovsdb table and updated the test covering all cases.  please review

Commit:
more apis to get/delete entries by vm name
